lighthouse-plugin-publisher-ads/utils/geometry.js

```diff
@@ -40,8 +40,39 @@ function boxViewableArea(clientRect, viewport) {
     (Math.min(bottom, innerHeight) - Math.max(top, 0));
 }
 
+/**
+ * Converts points (from a TraceEvent) to a ClientRect.
+ * @param {number[]} points
+ * @return {ClientRect}
+ */
+function toClientRect(points) {
+  return {
+    left: points[0],
+    top: points[1],
+    width: points[2],
+    height: points[3],
+    right: points[0] + points[2],
+    bottom: points[1] + points[3],
+  };
+}
+
+/**
+ * Checks ClientRect a overlaps ClientRect b. Note that this returns true for
```

Comment:
`Checks ClientRect a overlaps ClientRect b` -> `Checks if 2 ClientRects overlap`? Ordering doesn't matter here, right?

Reply:
Right. I initially had it that way, but then the name `overlaps` wouldn't be grammatical. It doesn't really matter since overlapping is a commutative property